### PR TITLE
Return null for unexpected HW intrinsic

### LINKED ISSUE #17815: HW intrinsics shouldn't assert for unsupported
For example, here:
https://github.com/dotnet/coreclr/blob/master/src/jit/hwintrinsicxarch.cpp#L816
As described in the comments for https://github.com/dotnet/coreclr/pull/17737, this can lead to an assert in the case where the user knows the path will not be taken. Such cases should simply return `nullptr` so that the IL method will be invoked.

### PR DESCRIPTION
**Pull request by CarolEidt**
Fix #17815

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib @tannergooding PTAL

File: src/jit/hwintrinsicxarch.cpp [hunk 1]
```diff
@@ -846,8 +846,11 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
         var_types               argType = TYP_UNKNOWN;
 
         assert(numArgs >= 0);
-        assert(HWIntrinsicInfo::lookupIns(intrinsic, baseType) != INS_invalid);
-        assert(simdSize == 32 || simdSize == 16);
+        if ((HWIntrinsicInfo::lookupIns(intrinsic, baseType) == INS_invalid) || ((simdSize != 32) && (simdSize == 16)))
```
Comment (tannergooding) [inline src/jit/hwintrinsicxarch.cpp]: Should be `simdSize != 16`?

Comment (CarolEidt) [reply src/jit/hwintrinsicxarch.cpp]: Oops!

**Comment by CarolEidt** [comment]
x64 Checked CoreFX test failures is a time-out, and the other is a flaky Arm64 test that can't possibly be related to this, since this only changes an xarch-only file.

**Comment by BruceForstall** [comment]
I disabled that arm64 test under gcstress with https://github.com/dotnet/coreclr/issues/24172

But if it repros under non-gcstress scenarios, we should expand the exclusion

**Comment by CarolEidt** [comment]
I commented on the `_il_relthread-race` test here: https://github.com/dotnet/coreclr/issues/23624, and Kusto shows a number of failures for that test.

**Pull request merged**